app/routes/recommend-to-reject.js

```diff
@@ -31,27 +30,23 @@ module.exports = {
       }
     },
     handler: async (request, h) => {
-      try {
-        const userRole = mapAuth(request)
-        if (!userRole.isRecommender && !userRole.isAdministrator) {
-          throw new Error('User must be a recommender or an admin')
-        }
-        await processStageActions(
-          request,
-          permissions.recommender,
-          stages.claimApproveReject,
-          stageExecutionActions.recommendToReject,
-          false
-        )
-        await crumbCache.generateNewCrumb(request, h)
-        if (request.payload.claimOrApplication === 'claim') {
-          return h.redirect(`/view-claim/${request.payload.reference}${request.payload?.returnPage && '?returnPage=' + request.payload?.returnPage}`)
-        } else {
-          return h.redirect(`/view-agreement/${request.payload.reference}?page=${request.payload.page}`)
-        }
-      } catch (error) {
-        console.error(`routes:recommend-to-reject: Error when processing request: ${error.message}`)
-        throw Boom.internal(error.message)
+      const userRole = mapAuth(request)
+      request.logger.setBindings({ userRole })
+      if (!userRole.isRecommender && !userRole.isAdministrator) {
```

Comment:
Might be slightly beyond the scope here, but I'm seeing this repeated a couple of times, maybe worth creating a util function somewhere along the lines of `userAllowedToRecommend(userRole) : boolean`

Reply:
Personally feel that's an abstraction too far, as the userRole object that's created gets checked in subtly different ways depending on the situation. 

Possibly the generation of the userRole object is something that could be done as part of the request lifecycle an made available on `request.app` for every request though. One to discuss outside the scope of this PR?

<img width="705" alt="Screenshot 2024-11-13 at 11 31 46 am" src="https://github.com/user-attachments/assets/aef9e001-70cd-4343-ad82-df753ed6075b">

Reply:
https://hapi.dev/api/?v=21.3.12#-requestapp

Reply:
We've got some work coming up to expand the functionality for changing statuses, might be good to have a look at this when that work is done